### PR TITLE
removing redundant package lists and sentence-transformers library

### PR DESCRIPTION
**Pull request by QuantumPlumber**
## Description

Removes the dependency: hugging face `sentence_transformer` package, that requires installation of `torch` and `cudnn`. This simplifies install and streamlines use of LLM endpoints.

## Type of change

(Please delete options that are not relevant)

- [x ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [x ]   Verification Steps:
 - Instantiated a RAG model using writer. No errors were thrown.

## Checklist:

- [ x] My code follows the style guidelines(PEP 8) of MindsDB.
- [ x] I have appropriately commented on my code, especially in complex areas.
- [ x] Necessary documentation updates are either made or tracked in issues.
- [ x] Relevant unit and integration tests are updated or added.

### CODE REVIEW

File: mindsdb/integrations/handlers/rag_handler/requirements.txt [hunk 1]
```diff
@@ -2,9 +2,8 @@
 faiss-cpu
 openai==1.6.1
 html2text
-sentence-transformers
 writerai~=1.1.0
 pydantic
-sentence-transformers
-faiss-cpu
 langchain-community  # for Writer LLM integration
+sentence-transformers # needed for HuggingFaceEmbeddings from langchain-community
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/rag_handler/requirements.txt:8]: @QuantumPlumber it fails without sentence-transformers. Adding back in.

Also updated unit test as they were failing as they hadn't been updated along with most recent code changes i.e. 'input_column' is non manidatory

 see https://github.com/langchain-ai/langchain/blob/b20c2640dac79551685b8aba095ebc6125df928c/libs/community/langchain_community/embeddings/huggingface.py#L65

File: mindsdb/integrations/handlers/rag_handler/settings.py [hunk 2]
```diff
@@ -175,6 +175,7 @@ def load_vector_store_client(
                 folder_path=self.config.persist_directory,
                 embeddings=self.config.embeddings_model,
                 index_name=self.config.collection_name,
+                allow_dangerous_deserialization=True
```
Comment (dusvyat) [inline mindsdb/integrations/handlers/rag_handler/settings.py:178]: Without this it fails with FAISS, seems to be a new parameter added in later versions of langchain for added security as FAISS is stored as a binary which gets deserialized on loading

File: mindsdb/integrations/handlers/rag_handler/requirements.txt (hunk 1 above)
Comment (QuantumPlumber) [reply mindsdb/integrations/handlers/rag_handler/requirements.txt:8]: This is only required if the user wants to use hugging face embeddings. I'm arguing for removing that support as it is not required for 3rd party LLM integration.

Comment (dusvyat) [reply mindsdb/integrations/handlers/rag_handler/requirements.txt:8]: hugging face embedings (sentence-transformers) is the only option for embeddings in rag_handler. Therefore sentence-transformers is required.

Comment (QuantumPlumber) [reply mindsdb/integrations/handlers/rag_handler/requirements.txt:8]: Oh.. so this will probably be cleaned up when RAG is consolidated.

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**